### PR TITLE
.Net changes for System.Text.Json support

### PR DESCRIPTION
**Pull request by andrueastman**
This PR highlights changes needed for raptor to support System.Text.Json

This PR is related to :
- https://github.com/microsoftgraph/microsoft-graph-explorer-api/pull/439
- https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/pull/382

### CODE REVIEW

**Review by zengin — APPROVED** [review]

File: msgraph-sdk-raptor-compiler-lib/msgraph-sdk-raptor-compiler-lib.csproj [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-    <PackageReference Include="Microsoft.Graph" Version="3.35.0" />
-    <PackageReference Include="Microsoft.Graph.Auth" Version="1.0.0-preview.7" />
-    <PackageReference Include="Microsoft.Graph.Beta" Version="0.49.0-preview" />
-    <PackageReference Include="Microsoft.Graph.Core" Version="1.*" />
+    <PackageReference Include="Microsoft.Graph" Version="4.0.0-preview.7" />
+    <PackageReference Include="Microsoft.Graph.Core" Version="2.0.0-preview.14" />
+    <PackageReference Include="Microsoft.Graph.Beta" Version="4.5.0-preview" />
     <PackageReference Include="Microsoft.Identity.Client" Version="4.33.0" />
-    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
+    <PackageReference Include="Microsoft.Graph.Auth" Version="1.0.0-preview.7" />
```
Comment (baywet) [inline msgraph-sdk-raptor-compiler-lib/msgraph-sdk-raptor-compiler-lib.csproj:17]: do we still have Microsoft.Graph.Auth with the upcoming version? I thought part of the goals were to leverage azure core/identity instead.

Comment (andrueastman) [reply msgraph-sdk-raptor-compiler-lib/msgraph-sdk-raptor-compiler-lib.csproj:17]: The `Microsoft.Graph.Auth` is used by the raptor test infrastructure to create a `IAuthenticationProvider` instance for running the tests. 
We could always refactor the raptor code as well to use the `Azure.Identity` token credentials when running the tests too.

**Review by MIchaelMainer — APPROVED** [review]

File: msgraph-sdk-raptor-compiler-lib/msgraph-sdk-raptor-compiler-lib.csproj (hunk 1 above)
Comment (baywet) [reply msgraph-sdk-raptor-compiler-lib/msgraph-sdk-raptor-compiler-lib.csproj:17]: maybe not right now but I thought part of the plan was to deprecate that package once we release the newer version of the dotnet SDK.
Having our integration tests depend on a deprecated package would not be optimal, maybe create an issue for post release cleanup?

Comment (andrueastman) [reply msgraph-sdk-raptor-compiler-lib/msgraph-sdk-raptor-compiler-lib.csproj:17]: Agreed. This is now created here. https://github.com/microsoftgraph/msgraph-sdk-raptor/issues/307

Comment (baywet) [reply msgraph-sdk-raptor-compiler-lib/msgraph-sdk-raptor-compiler-lib.csproj:17]: thanks for taking the time to clear the confusion on my side and logging it. Resolving.

**Review by baywet — APPROVED** [review]

**Pull request merged**